### ISSUE TITLE
biber > 1.9 dies (?) with error 255

### ISSUE THREAD

**Issue by cbm755** [plk/biber#40]
Using recent git commit d9d65aa and also tested with 1.9 (+1 commit), biber seems to stop without a clear error message.  I'm not a heavy biber user (this is my only document I've used it for) so I'm not much use a debugging this!

Anyway, the output:

```
biber report
INFO - This is Biber 2.0 (beta)
INFO - Logfile is 'report.blg'
INFO - Reading 'report.bcf'
WARN - Warning: Found biblatex control file version 2.5, expected version 2.6
INFO - Found 50 citekeys in bib section 0
INFO - Found 8 citekeys in bib section 1
INFO - Found 0 citekeys in bib section 0
Use of uninitialized value $name in lc at /usr/share/perl5/Biber/SortList.pm line 87.
Use of uninitialized value $lssn in concatenation (.) or string at /usr/share/perl5/Biber.pm line 682.
Use of uninitialized value $lname in concatenation (.) or string at /usr/share/perl5/Biber.pm line 682.
Use of uninitialized value $name in string eq at /usr/share/perl5/Biber/SortLists.pm line 77.
Use of uninitialized value in string eq at /usr/share/perl5/Biber/SortLists.pm line 77.
Use of uninitialized value $ssn in string eq at /usr/share/perl5/Biber/SortLists.pm line 77.
Use of uninitialized value $lname in concatenation (.) or string at /usr/share/perl5/Biber.pm line 661.
Use of uninitialized value $lssn in concatenation (.) or string at /usr/share/perl5/Biber.pm line 661.
Use of uninitialized value $name in string eq at /usr/share/perl5/Biber/SortLists.pm line 77.
Use of uninitialized value in string eq at /usr/share/perl5/Biber/SortLists.pm line 77.
Use of uninitialized value $ssn in string eq at /usr/share/perl5/Biber/SortLists.pm line 77.
Use of uninitialized value $name in lc at /usr/share/perl5/Biber/SortList.pm line 87.
Use of uninitialized value $lssn in concatenation (.) or string at /usr/share/perl5/Biber.pm line 682.
Use of uninitialized value $lname in concatenation (.) or string at /usr/share/perl5/Biber.pm line 682.
INFO - Processing section 0
INFO - Looking for bibtex format file 'cbm-pub.bib' for section 0
INFO - Decoding LaTeX character macros into UTF-8
INFO - Found BibTeX data source '/home/cbm/work/texinputs/cbm-pub.bib'
make: *** [all] Error 255
```

I can't tell what is the error message here, is it the biblatex warning?

Here's the output from biber 1.8 (which works fine)

```
biber report
INFO - This is Biber 1.8
INFO - Logfile is 'report.blg'
INFO - Reading 'report.bcf'
INFO - Found 50 citekeys in bib section 0
INFO - Found 8 citekeys in bib section 1
INFO - Found 0 citekeys in bib section 0
INFO - Processing section 0
INFO - Looking for bibtex format file 'cbm-pub.bib' for section 0
INFO - Found BibTeX data source '/home/cbm/work/texinputs/cbm-pub.bib'
INFO - Looking for bibtex format file 'cbm-students.bib' for section 0
INFO - Found BibTeX data source '/home/cbm/work/texinputs/cbm-students.bib'
INFO - Looking for bibtex format file 'cbm-talks.bib' for section 0
INFO - Found BibTeX data source '/home/cbm/work/texinputs/cbm-talks.bib'
INFO - Overriding locale 'en_US.UTF-8' default tailoring 'variable = shifted' with 'variable = non-ignorable'
INFO - Sorting 'entry' list 'none' keys
INFO - No sort tailoring available for locale 'en_US.UTF-8'
INFO - Processing section 1
INFO - Looking for bibtex format file 'cbm-pub.bib' for section 1
INFO - Looking for bibtex format file 'cbm-students.bib' for section 1
INFO - Overriding locale 'en_US.UTF-8' default tailoring 'variable = shifted' with 'variable = non-ignorable'
INFO - Sorting 'entry' list 'none' keys
INFO - No sort tailoring available for locale 'en_US.UTF-8'
INFO - Writing 'report.bbl' with encoding 'ascii'
INFO - Output to report.bbl
```

**Comment by espinozahg** [comment]
I have somewhat similar error in Fedora 21 workstation 64 bit. I have TeX Live 2014 from standard/updates repositories and biber from https://copr.fedoraproject.org/coprs/cbm/Biber/
I think another key message is:
    INFO - This is Biber 2.0 (beta)
    which I get too.
Additionally,
    biber -v
    outputs:
    biber version: 2.0 (beta)
Shouldn't it be 1.9 instead of 2.0 (beta)? Because the standard pdflatex I have installed writes a .bcf in format version 2.5 and that can be understood by biber 1.9. On the other hand, biber 2.0 expects a .bcf in format version 2.6 which is not there.
So, long story short: we probably need something like 
    http://copr.fedoraproject.org/coprs/cbm/biber-1.9
    for Fedora 21

**Comment by cbm755** [comment]
@espinozahg, you can use this for now: https://copr.fedoraproject.org/coprs/cbm/f21-biber-1.8

**Comment by espinozahg** [comment]
thanks @cbm755 . I tried f21-biber-1.8 now and works correctly.
(fedora 21 64 bit workstation with up-to-date package texlive)

**Comment by espinozahg** [comment]
@cbm755 :
After installation cbm/f21-biber-1.8 worked without any warning as I mentioned before, but now It is not working properly (after reboot and clean environment in terminal). I haven't had the time to see what is going on. I switched to cbm/Biber which outputs lots of warnings (perl related and biber/biblatex version compatibility related) but produces a visually OK pdf file with pdflatex.

**Comment by cbm755** [comment]
Hmmm could you get us an error msg?

**Comment by espinozahg** [comment]
I uninstalled biber, disabled cbm/Biber, enabled cbm/f21-biber-1.8, installed biber again and rebooted just in case but I cannot reproduce the error I got before with cbm/f21-biber-1.8
I could post the errors/warnings with cbm/Biber but I don't think they are needed, are they?
I will continue using cbm/f21-biber-1.8 and report if any error/warning appears again.

**Comment by cbm755** [comment]
> I could post the errors/warnings with cbm/Biber but I don't think they are needed, are they?

That's right, its deprecated, no reason it should work.

**Comment by espinozahg** [comment]
I'm back again. I have one file set that doesn't work with biber from cbm/f21-biber-1.8 . The file set has one .bcf file and one .bib file
Executing
biber thesis
I get no error or warning (either on screen or on the .blg file), but the .bbl file is 0 bytes.
I've tried biber with the following options to see what is going on, but still, no error or warning, biber just dies with exit code 1 (echo $?). The options are:  --nodieonerror --decodecharsset=full --trace
The files are:
https://sites.google.com/site/espinozahg/stuff/aux-references.bib?attredirects=0&d=1
https://sites.google.com/site/espinozahg/stuff/thesis.bcf?attredirects=0&d=1
I have to mention that this file set works with TeX Live 2014 under Ubuntu 12.04

**Comment by cbm755** [comment]
Perhaps you should file a new issue?  When you do, you'd better clearly state biber versions used on each system.  Also, narrow down to a minimal nonworking example.
